File: lib/to_mdx/parsers/open_document/text/list_item.rb
# frozen_string_literal: true

require_relative "../base"

module ToMdx
  module Parsers
    module Text
      class ListItem < Base
        def to_mdx
          "<li>#{contentful_children.map(&:to_mdx).join}</li>"
        end

        def to_s
          "- #{contentful_children.map(&:to_s).map(&:chomp).join(" ")}"
        end

        register_for "text:list-item"
      end
    end
  end
end

File: src/parsers/open-document/text/list_item.ts
import { BaseElement } from "../../base_element.ts";

export class ListItem extends BaseElement {
  toMdx(): string {
    const content = this.contentfulChildren()
      .map((child) => child.toMdx())
      .join("");
    return `<li>${content}</li>`;
  }

  toString(): string {
    const content = this.contentfulChildren()
      .map((child) => child.toString())
      .map((str) => str.replace(/\n$/, "")) // chomp equivalent
      .join(" ");
    return `- ${content}`;
  }

  static {
    BaseElement.registerFor.call(this, "text:list-item");
  }
}
